src/Native/System.Native/pal_networking.cpp

```diff
@@ -2338,10 +2344,20 @@ static Error TryChangeSocketEventRegistrationInner(
                0,
                0,
                reinterpret_cast<void*>(data));
+#else
+         EV_SET(&events[i++],
+               static_cast<uint64_t>(socket),
+               EVFILT_READ,
+               (newEvents & PAL_SA_READ) == 0 ? RemoveFlags : AddFlags,
+               0,
+               0,
+               data);
+#endif
```

Comment:
Rather than duplicating each of these whole blocks simply to pass either ```data``` or ```reinterpret_cast<void*>(data)``` (and similarly for the other cases below), did you consider just having a macro that expanded to the right one?  Then we wouldn't have to have all of this duplicated code.